aws/sra-test/integration-tests/aws-sdk-s3/tests/sra_test.rs

```diff
@@ -3,280 +3,98 @@
-                    let input = context.input()?;
-                    let input = input
-                        .downcast_ref::<ListObjectsV2Input>()
-                        .ok_or_else(|| "failed to downcast to ListObjectsV2Input")?;
-                    let mut params_builder = cfg
-                        .get::<aws_sdk_s3::endpoint::ParamsBuilder>()
-                        .ok_or_else(|| "missing endpoint params builder")?
-                        .clone();
-                    params_builder = params_builder.set_bucket(input.bucket.clone());
-                    cfg.put(params_builder);
-
-                    Ok(())
-                }
-            }
-
-            #[derive(Debug)]
-            struct ListObjectsV2EndpointParamsFinalizerInterceptor;
-            impl Interceptor<HttpRequest, HttpResponse> for ListObjectsV2EndpointParamsFinalizerInterceptor {
-                fn read_before_execution(
-                    &self,
-                    _context: &InterceptorContext<HttpRequest, HttpResponse>,
-                    cfg: &mut ConfigBag,
-                ) -> Result<(), BoxError> {
-                    let params_builder = cfg
-                        .get::<aws_sdk_s3::endpoint::ParamsBuilder>()
-                        .ok_or_else(|| "missing endpoint params builder")?
-                        .clone();
-                    let params = params_builder.build().map_err(|err| {
-                        ContextAttachedError::new("endpoint params could not be built", err)
-                    })?;
-                    cfg.put(
-                        aws_smithy_runtime_api::client::orchestrator::EndpointResolverParams::new(
-                            params,
-                        ),
-                    );
-
-                    Ok(())
-                }
-            }
-
-            cfg.get::<Interceptors<HttpRequest, HttpResponse>>()
-                .expect("interceptors set")
-                .register_operation_interceptor(
-                    Arc::new(ListObjectsV2EndpointParamsInterceptor) as _
-                )
-                .register_operation_interceptor(Arc::new(
-                    ListObjectsV2EndpointParamsFinalizerInterceptor,
-                ) as _);
-            Ok(())
-        }
+struct FixupPlugin {
+    client: Client,
+    timestamp: SystemTime,
+}
+impl RuntimePlugin for FixupPlugin {
+    fn configure(
+        &self,
+        cfg: &mut ConfigBag,
+    ) -> Result<(), aws_smithy_runtime_api::client::runtime_plugin::BoxError> {
+        let params_builder = Params::builder()
+            .set_region(self.client.conf().region().map(|c| c.as_ref().to_string()))
+            .bucket("test-bucket");
+
+        cfg.put(params_builder);
+        cfg.set_request_time(RequestTime::new(self.timestamp.clone()));
+        cfg.put(AwsUserAgent::for_tests());
+        cfg.put(InvocationId::for_tests());
+        Ok(())
     }
-
-    let conn = TestConnection::new(vec![(
```

Comment:
I would prefer to keep this test using a pre-recorded request/response.

Reply:
it is still prerecorded but I changed it to use DVR, will make it easier to add more